src/hooks/useGasInfo.ts

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from 'react'
+import { useActiveWeb3React } from '.'
+import { ChainId } from '@swapr/sdk'
+import { INFURA_PROJECT_ID } from '../connectors'
+
+const gasInfoChainUrls: {
+  [chainId: number]: { url: string; body?: any; objectDestrucutre?: any; deepDestrucuture?: string }
+} = {
+  [ChainId.MAINNET]: {
+    url: 'http://ethgas.watch/api/gas',
+    objectDestrucutre: ['normal', 'fast', 'slow']
+  },
+  [ChainId.XDAI]: {
+    url: 'https://blockscout.com/xdai/mainnet/api/v1/gas-price-oracle',
+    objectDestrucutre: ['average', 'fast', 'slow']
+  },
+  [ChainId.ARBITRUM_ONE]: {
+    url: `https://arbitrum-mainnet.infura.io/v3/${INFURA_PROJECT_ID}`,
+    body: {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        jsonrpc: '2.0',
+        method: 'eth_gasPrice',
+        params: [],
+        id: 1
+      })
+    }
+  }
+}
+export function useGasInfo(): { loading: boolean; gas: { fast: number; normal: number; slow: number } } {
+  const { chainId } = useActiveWeb3React()
+  const [loading, setLoading] = useState<boolean>(true)
+  const [gas, setGas] = useState<{ fast: number; normal: number; slow: number }>({ normal: 0, fast: 0, slow: 0 })
```

Comment:
`{ fast: number; normal: number; slow: number }` can be extracted into a single type.